### PR TITLE
Stream with utf8 filename

### LINKED ISSUE #790: response headers support custom encoding
Hi~ When i try to return a `FileResponse` with Chinese filename, i got such error:

```
UnicodeEncodeError: 'latin-1' codec can't encode character '\u6708' in position 32: ordinal not in range(256)
```

Then i found all header items will be encode with `latin-1`, after i change it manually to `utf-8`, my problem was solved.

e.g.

- responses.Response.init_headers, line 63
```python
raw_headers = [
    (k.lower().encode("latin-1"), v.encode("latin-1"))
    for k, v in headers.items()
]
```

So may be it should encode with `self.charset` or give a custom parameter so that i can choose encoding to `utf-8`?

Thx for the great project~

### PR DESCRIPTION
**Pull request by Yasu-umi**
Fixes: https://github.com/encode/starlette/issues/790#issuecomment-667881654

This adds support filename for StreamingResponse  same as FileResponse.

### CODE REVIEW

**Comment by addisonElliott** [comment]
Running across this bug. I am definitely interested in getting this merged!

File: starlette/responses.py [hunk 1]
```diff
@@ -34,6 +34,18 @@ def guess_type(
     return mimetypes_guess_type(url, strict)
 
 
+# Build Content Disposition Header from filename
```
Comment (Kludex) [inline starlette/responses.py]: No need, the function name is pretty clear. :)

```suggestion
```

**Review by Kludex — COMMENTED** [review]
I'm looking at the history, and it looks like conceptually `StreamingResponse` shouldn't be accepting `filename`.

@tomchristie Can you confirm?

**Comment by adriangb** [comment]
> I'm looking at the history, and it looks like conceptually `StreamingResponse` shouldn't be accepting `filename`.

Would this work for you @Yasu-umi ?

```python
import os
from tempfile import TemporaryDirectory

from starlette.responses import FileResponse
from starlette.testclient import TestClient

def test_streaming_response_with_japanese_filename() -> None:
    filename = "こんにちは.txt"  # "Hello.txt" in Japanese
    content = b"1, 2, 3, 4, 5"
    with TemporaryDirectory() as dirname:
        filepath = os.path.join(dirname, filename)
        with open(filepath, mode="wb") as f:
            f.write(content)
        client = TestClient(FileResponse(filepath, filename=filename))
        response = client.get("/")
    expected_disposition = (
        "attachment; filename*=utf-8''%E3%81%93%E3%82%93%E3%81%AB%E3%81%A1%E3%81%AF.txt"
    )
    assert response.status_code == 200
    assert response.content == content
    assert response.headers["content-disposition"] == expected_disposition
```

**Comment by Yasu-umi** [comment]
@adriangb thanks for example!
I solved this by below codes (just overwrite header).
```python
def create_file_response(*, stream: BytesIO, filename: str, **kwargs: Any) -> StreamingResponse:
  res = StreamingResponse(stream, **kwargs)
  res.raw_headers.append((
    b'content-disposition',
    f"attachment; filename*=utf-8''{quote(filename)}".encode('latin-1'),
  ))
  return res
```
As Kludex taught me, if conceptually `StreamingResponse` shouldn't be accepting `filename`, should I close this pull req & issue ?

**Comment by adriangb** [comment]
Yes, please close it

**Pull request closed without merging**